libs/barista-components/chart/src/tooltip/chart-tooltip.ts

```diff
@@ -117,7 +99,13 @@ export class DtChartTooltip implements OnDestroy {
       const positionStrategy = this._overlay
         .position()
         .flexibleConnectedTo(
-          this._getTooltipPosition(data, parentChart, plotBackgroundInfo),
+          this.customPositioner?.positionFunction !== undefined
+            ? this.customPositioner.positionFunction(
+                data,
+                parentChart,
+                plotBackgroundInfo,
+              )
+            : getDefaultTooltipPosition(data, parentChart, plotBackgroundInfo),
```

Comment:
Nitpick: 
as this is not an assignment, I do prefer non ternary if checks in this case. I believe this is more readable: 
```
if (this.customPositioner?.positionFunction !== undefined) {
  this.customPositioner.positionFunction(
    data,
    parentChart,
    plotBackgroundInfo,
  )
} else { 
  getDefaultTooltipPosition(data, parentChart, plotBackgroundInfo)
}

Reply:
It is an assignment but the ternary is used for computing directly the parameter for flexibleConnectedTo.
I can change it to something like this if you prefer:

`
let positioner = this.customPositioner?.positionFunction !== undefined
  ? this.customPositioner.positionFunction(data,  parentChart, plotBackgroundInfo)
 : getDefaultTooltipPosition(data, parentChart, plotBackgroundInfo);

const positionStrategy = this._overlay
        .position()
        .flexibleConnectedTo(positioner, ..... );
`